PetrolEngine.cpp

```diff
@@ -15,6 +16,47 @@ void PetrolEngine::changeGear(int gear)
     // TODO: Add checking if gear is between -1 and gears_
     // -1 is for REAR
     // 0 is for NEUTRAL
-    currentGear_ = gear;
+    
     std::cout << __FUNCTION__ << std::endl;
+    try
+    {
+        if (gear >= -1 && gear <= gears_) 
+        {
+            try
+            { 
+                if (currentGear_ == -1 && gear != 1)
+                {   
+//                  throw std::logic_error("Can't change gear from rear to higher than 1");
+                    throw 0;
```

Comment:
Rzucanie liczbami to nie jest dobra praktyka. Nikt nie wie co taka liczba oznacza i z punktu widzenia programu rozróżniane są typy wyjątków, a tutaj będzie int w każdym przypadku, więc to bez różnicy jaką wartość rzucimy. Ale to co jest zakomentowane wygląda już znacznie sensowniej. Używanie wyjątków z biblioteki standardowej jest ok.